coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,395 @@
+        ' function', (0, 1, 2), default=1)
+
+
+@BlankLine.subaspect
+class BlankLineAfterClass:
+    """
+    Blank lines after classes refer to the blank lines found after classes'
+    definitions.
+    """
+    class docs:
+        example = """
+        class SomeClass:
+            def __init__(self):
+                raise RuntimeError('Never instantiate this class')
+
+
+        def func():
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific number of blank lines after every classes'
+        definitions declarations improves on the readability of the code.
+        """
+        fix_suggestions = """
+        """
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class'
+        'definition.', (0, 1, 2, 3), default=2)
+
+
+@BlankLine.subaspect
+class NewlineAtEOF:
+    """
+    A newline character (usually '\n', aka CR or CRLF) at  found at the end of
+    file.
+    """
+    class docs:
+        example = """
+        def do_nothing():
+            pass\n
+        """
+        example_language = 'python'
+        importance_reason = """
+        A text file, under unix, consists of a series of lines, each of which
+        ends with a newline character (\\n). A file that is not empty and does
+        not end with a newline is therefore not a text file.
+
+        It's not just bad style, it can lead to unexpected behavior, utilities
+        that are supposed to operate on text files may not cope well with files
+        that don't end with a newline.
+        """
+        fix_suggestions = """
+        `NewlineAtEOF` issues can be fixed by simply adding a newline at the
+        end of the file.
+        """
+    enforce_newline_at_EOF = Taste[bool](
+        'If ``False``, enforce a newline at End Of File.',
+        (True, False), default=True)
+
+
+@Spacing.subaspect
+class SpacesAroundOperator:
+    """
+    Spacing around operators.
+    """
+    class docs:
+        example = """
+        def f(a):
+            return 37+-+a[42-x :  y**3]
+        """
+        example_language = 'python'
+        importance_reason = """
+        Having a specific and reasonable number of whitespace (blank) around
+        operators improves on the readability of the code.
+        """
+        fix_suggestions = """
+        `SpacesAroundOperator` issues can be fixed by simply specifying (and
+        using) a number of whitespace to be use after each operator.
+        """
+    spaces_around_operators = Taste[int](
+        'Represents the number of space to be used around operators.',
+        (0, 1), default=1)
+    spaces_before_colon = Taste[int](
+        'Represents the number of blank spaces before colons.',
+        (0, 1), default=0)
+    spaces_after_colon = Taste[int](
+        'Represents the number of blank spaces after colons.',
+        (0, 1), default=1)
```

Comment:
maybe not now but this'll have to be split up because we might have a bear that only checks one of those and we might have users that only want to check one of those